lib/components/private/withTheme.tsx

```diff
@@ -34,12 +37,16 @@ const withTheme = <P extends WithThemeProps>(Component: ComponentType<P>) => {
             }
 
             return (
-              <Component {...this.props} theme={this.props.theme || theme} />
+              <OriginalComponent
+                theme={theme}
+                {...this.props as any} // Workaround for https://github.com/Microsoft/TypeScript/issues/28748
```

Comment:
Did you change the order of these props deliberately? The `theme` should come after the spreading of props right?